cellprofiler/modules/measureradialentropy.py

```diff
@@ -0,0 +1,205 @@
+            "Input number of bins", 6, minval=3, maxval=60,
+            doc="""Number of radial bins to divide your object into.  The minimum number
+            of bins allowed is 3, the maximum number is 60.""" % globals())
+
+        self.intensity_measurement=cps.Choice(
+            "Which intensity measurement should be used?", ['Mean','Median'], value='Mean',doc="""
+            Whether each wedge's mean or median intensity should be used to calculate the entropy.""" % globals())
+
+    def settings(self):
+        return [self.input_image_name, self.input_object_name,
+                self.intensity_measurement, self.bin_number]
+
+
+    def run(self, workspace):
+        #Import the workspace and the current measurements
+        meas = workspace.measurements
+        assert isinstance(meas, cpmeas.Measurements)
+
+        statistics = [["Entropy"]]
+
+        workspace.display_data.statistics = statistics
+
+        #Import the settings
+        input_image_name = self.input_image_name.value
+        input_object_name = self.input_object_name.value
+        metric = self.intensity_measurement.value
+        bins = self.bin_number.value
+
+        image_set = workspace.image_set
+
+        input_image = image_set.get_image(input_image_name,
+                                          must_be_grayscale=True)
+        #Read out the pixel data
+        pixels = input_image.pixel_data
+
+        object_set = workspace.object_set
+        assert isinstance(object_set, cpo.ObjectSet)
+
+        objects = object_set.get_objects(input_object_name)
+        labels = objects.segmented
+
+        indexes = objects.indices
+        #Calculate the center of the objects- I'm guessing there's a better way to do this but this was already here
+        centers, radius = minimum_enclosing_circle(labels, indexes)
+
+        feature = self.get_measurement_name(input_image_name,metric,bins)
+        #Do the actual calculation
+        entropy=self.slice_and_measure_intensity(pixels,labels,indexes,centers,metric,bins)
+        #Add the measurement back into the workspace
+        meas.add_measurement(input_object_name,feature,entropy)
+
+        emean = np.mean(entropy)
+        statistics.append([feature, emean])
+
+
+    ################################
+    #
+    # DISPLAY
+    #
+    def display(self, workspace, figure=None):
+        statistics = workspace.display_data.statistics
+        if figure is None:
+            figure = workspace.create_or_find_figure(subplots=(1, 1,))
+        else:
+            figure.set_subplots((1, 1))
+        figure.subplot_table(0, 0, statistics)
+
+
+    def slice_and_measure_intensity(self, pixels, labels, indexes,centers,metric,nbins):
+        '''For each object, iterate over the pixels that make up the object, assign them to a bin,
+        then call calculate_entropy and return it to run.  I'm sure whatever I did here will make
+        Allen and Claire weep tears of bad-code-sadness'''
```

Comment:
LOL! This is really good! My tears are tears of joy! I think the only change I’d make from a structural perspective is to use NumPy vector operations rather than iteration for the performance benefit.